staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker_test.go

```diff
@@ -219,34 +218,6 @@ func TestSupportsRequestWatchProgress(t *testing.T) {
 					{Version: "3.5.13", Endpoint: "localhost:2393"}},
 					expectedResult: false}},
 		},
-		{
-			testName: "Enabled - error on first client, enabled success on second client",
-			rounds: []testCase{
-				{endpointsVersion: []mockEndpointVersion{
-					{Version: "3.6.0", Endpoint: "localhost:2390", Error: fmt.Errorf("some error")}},
-					expectedResult: false,
-					expectedError:  fmt.Errorf("failed checking etcd version, endpoint: %q: %w", "localhost:2390", fmt.Errorf("some error")),
-				},
-				{endpointsVersion: []mockEndpointVersion{
-					{Version: "3.5.14", Endpoint: "localhost:2391"}},
-					expectedResult: true}},
-		},
-		{
-			testName: "Disabled - enabled success on first client, error on second client, disabled success on third client",
```

Comment:
Also with changing the way `CheckClient` works as async, im not sure how to check these tests. Maybe a `chan` ?
Or maybe we remove these tests?

Reply:
We should not remove those tests, we need just make the mock smarter. Mock can provide a channel that you can wait on that gets updated when it receives a request.

Reply:
Basically:
- change checkClient to be synchronous
- in the place where you call `checkClient()` switch to `go checkClient()`
- switch the tests to use internal function instead of public one